test/common_nn.py

```diff
@@ -680,7 +682,7 @@ def _analytical_jacobian(self, module, input, jacobian_input=True, jacobian_para
 
             if jacobian_input:
                 for jacobian_x, d_x in zip(flat_jacobian_input, iter_tensors(d_input)):
-                    jacobian_x[:, i] = d_x
+                    jacobian_x[:, i] = d_x.contiguous().view(-1)
```

Comment:
was there a case that broke here?

Reply:
Yes, the assignment didn't obey broadcasting rules.

Reply:
Tensor `__setitem__` didn't fully adhere to broadcasting rules. `Variable.__setitem__` does

Reply:
I'm still missing something -- do we actually need broadcasting here or can we assert that the shapes are the same?

Reply:
We need the `.view()` call. The jacobian is always 2d (`n_inputs x n_outputs`). The gradient d_x can be any shape so we need to view it as 1-d